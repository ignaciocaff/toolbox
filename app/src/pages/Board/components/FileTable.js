import React from "react";
import Table from "react-bootstrap/Table";

const FileTable = ({ data }) => {
  if (!data || data.length === 0) {
    return null;
  }

  return (
    <Table className="mt-2" responsive bordered hover>
      <thead>
        <tr>
          <th>File name</th>
          <th>Text</th>
          <th>Number</th>
          <th>Hex</th>
        </tr>
      </thead>
      <tbody>
        {data.map((file, index) => (
          <React.Fragment key={index}>
            <tr>
              <td className="align-middle" rowSpan={file?.lines?.length + 1}>
                {file.file}
              </td>
            </tr>
            {file.lines.map((line, lineIndex) => (
              <tr key={lineIndex}>
                <td>{line.text}</td>
                <td>{line.number?.toLocaleString()?.replace(/[.,]/g, '')}</td>
                <td>{line.hex}</td>
              </tr>
            ))}
          </React.Fragment>
        ))}
      </tbody>
    </Table>
  );
};

export default FileTable;
