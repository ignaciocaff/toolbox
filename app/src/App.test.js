import renderer from "react-test-renderer";
import FileTable from "./pages/Board/components/FileTable";

it("DeberÃ­a renderizar correctamente la grilla para un archivo", () => {
  const testData = [
    {
      file: "test2.csv",
      lines: [
        {
          text: "dfTwtPGBOHF",
          number: 374837,
          hex: "282536d30b5b88249402318f14de59a5",
        },
      ],
    }
  ];
  const tree = renderer
    .create(<FileTable data={testData}></FileTable>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});