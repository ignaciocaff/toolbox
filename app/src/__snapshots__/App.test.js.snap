// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deber√≠a renderizar correctamente la grilla para un archivo 1`] = `
<div
  className="table-responsive"
>
  <table
    className="mt-2 table table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          File name
        </th>
        <th>
          Text
        </th>
        <th>
          Number
        </th>
        <th>
          Hex
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="align-middle"
          rowSpan={2}
        >
          test2.csv
        </td>
      </tr>
      <tr>
        <td>
          dfTwtPGBOHF
        </td>
        <td>
          374837
        </td>
        <td>
          282536d30b5b88249402318f14de59a5
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
